cpu/nrf51/periph/pwm.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_nrf51
+ * @ingroup     drivers_periph_pwm
+ * @{
+ *
+ * @file
+ * @brief       Low-level PWM driver implementation
+*
+ * @note        This driver initializes a duty cycle of '1'
+ *
+ * @author      Semjon Kerner <semjon.kerner@fu-berlin.de>
+ * @}
+ */
+
+#include <stdint.h>
+#include <string.h>
+#include <math.h>
+
+#include "periph/gpio.h"
+#include "periph/pwm.h"
+#include "vectors_cortexm.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#define PWM_PS_MAX          (9U)
+#define PWM_PPI_CHANNELS    (((1U) << PWM_PPI_A) | ((1U) << PWM_PPI_B))
+#ifndef PWM_PERCENT_VAL
+    #define PWM_PERCENT_VAL (1U)
+#endif
+
+static const uint32_t divtable[11] = {
+    (CLOCK_CORECLOCK >> 0),
+    (CLOCK_CORECLOCK >> 1),
+    (CLOCK_CORECLOCK >> 2),
+    (CLOCK_CORECLOCK >> 3),
+    (CLOCK_CORECLOCK >> 4),
+    (CLOCK_CORECLOCK >> 5),
+    (CLOCK_CORECLOCK >> 6),
+    (CLOCK_CORECLOCK >> 7),
+    (CLOCK_CORECLOCK >> 8),
+    (CLOCK_CORECLOCK >> 9),
+    (CLOCK_CORECLOCK >> 10),
```

Comment:
the maximum allowed prescaler value is `9`, right? So we have 10 possible values -> this array should have only 10 members, correct?